sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -240,17 +241,16 @@ public Mono<RxDocumentServiceResponse> performRequestInternal(RxDocumentServiceR
                 httpHeaders,
                 contentAsByteArray);
 
-            Duration responseTimeout = Duration.ofSeconds(Configs.getHttpResponseTimeoutInSeconds());
-            if (OperationType.QueryPlan.equals(request.getOperationType())) {
-                responseTimeout = Duration.ofSeconds(Configs.getQueryPlanResponseTimeoutInSeconds());
-            } else if (request.isAddressRefresh()) {
-                responseTimeout = Duration.ofSeconds(Configs.getAddressRefreshResponseTimeoutInSeconds());
+            // TODO: Clean this code up later
+            if (request.getResponseTimeout() == null) {
+                request.setResponseTimeout(HttpTimeoutPolicy.getTimeoutPolicy(request).
+                    getTimeoutAndDelaysList().get(0).getResponseTimeout());
```

Comment:
This can cause IndexOutOfBound exception, worth checking for list is not empty before dereferencing it.

Reply:
Fixed it.